### PR TITLE
Add Popover API examples

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

The [Popover API](https://html.spec.whatwg.org/multipage/popover.html) is going to be available in Chrome 114. I've had this confirmed by the relevant engineers, and it is also reflected by the platform status entry: https://chromestatus.com/feature/5463833265045504.

This PR adds several examples to the repo, to demonstrate different aspects of the Popover API. There might be some small changes going forward, but what is here is fairly complete as it stands.

See my [research document](https://docs.google.com/document/d/1ryC_4hvkLom0E7aaNnAB82_Ki5Rb_cKgrOJGSBU9fuw/edit#) for further details of what has been added in this PR.

cc @keithamus

### CODE REVIEW

**Comment by teoli2003** [comment]
Can you run Prettier on these files (`npx prettier -w *` on the popover_api directory)? This will fix all indentation problems as well as spacing (like `if(...)` instead of `if (...)`). There may be a few extra nits afterward.

**Review by keithamus — APPROVED** [review]
Generally LGTM :+1:

File: README.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ Code examples that accompany various MDN DOM and Web API documentation pages.
 
 * The "pointer-lock" directory contains a simple demo to show usage of the Pointer Lock API. You can find more explanation of how the demo works at the main MDN [Pointer Lock API](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_Lock_API) page. [Run the demo live](https://mdn.github.io/dom-examples/pointer-lock/).
 
+* The "popover-api" directory is for examples and demos of the [Popover API](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API) standard. Go to the[Popover API demo index](https://mdn.github.io/dom-examples/popover-api/) to see what's available.
```
Comment (teoli2003) [inline README.md:38]:
```suggestion
* The "popover-api" directory is for examples and demos of the [Popover API](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API) standard. Go to the [Popover API demo index](https://mdn.github.io/dom-examples/popover-api/) to see what's available.
```

File: popover-api/nested-popovers/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+// Event to make the subpopover keyboard-accessible
+
+subcontainer.addEventListener("focusin", () => {
+  subpopover.showPopover();
+});
+
+// Events to hide the popover menus when an option is selected in either of them
+
+mainpopover.addEventListener("click", () => {
```
Comment (teoli2003) [inline popover-api/nested-popovers/index.js:25]: Nit (spacing)
```suggestion
// Events to show/hide the subpopover when the mouse moves over and out
subcontainer.addEventListener("mouseover", () => {
  subpopover.showPopover();
});

subcontainer.addEventListener("mouseout", () => {
  if (subpopover.matches(":popover-open")) {
    subpopover.hidePopover();
  }
});

// Event to make the subpopover keyboard-accessible
subcontainer.addEventListener("focusin", () => {
  subpopover.showPopover();
});

// Events to hide the popover menus when an option is selected in either of them
mainpopover.addEventListener("click", () => {
```

**Review by teoli2003 — APPROVED** [review]

**Comment by chrisdavidmills** [comment]
@teoli2003 cheers!

**Pull request merged**